### PR TITLE
jobs/{build,build-arch,bump-lockfile}: start using kola() kolaTestIso()

### PR DESCRIPTION
**Pull request by dustymabe**
With recent updates to coreos-ci-lib [1] we can now using the kola() and kolaTestIso() functions even for multi-arch. This enables us to share code for kola tests across the three places that had heavily duplicated code in the past.

[1] https://github.com/coreos/coreos-ci-lib/pull/120

### CODE REVIEW

**Comment by dustymabe** [comment]
Requires https://github.com/coreos/coreos-assembler/pull/3124

**Comment by dustymabe** [comment]
Requires https://github.com/coreos/coreos-ci-lib/pull/120

**Comment by dustymabe** [comment]
added a second commit to switch the kola jobs from `fcosKola()` to just `kola()`

File: jobs/build-arch.Jenkinsfile [hunk 1]
```diff
@@ -161,7 +161,9 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
         // We set the session to time out after 4h. This essentially
         // performs garbage collection on the remote if we fail to clean up.
         pipeutils.withPodmanRemoteArchBuilder(arch: basearch) {
-        def session = shwrapCapture("cosa remote-session create --image ${image} --expiration 4h")
+        def session = shwrapCapture("""
+        cosa remote-session create --image ${image} --expiration 4h --workdir ${env.WORKSPACE}
+        """)
```
Comment (jlebon) [inline jobs/build-arch.Jenkinsfile:166]: This feels like it should be a separate commit with its own motivation.

File: jobs/build-arch.Jenkinsfile [hunk 2]
```diff
@@ -331,77 +333,14 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
...
-            }
-        }
-
-        // process this batch
-        parallel parallelruns
-
+        // Run Kola Tests
+        def n = 4 // VMs are 2G each and arch builders have approx 32G
+        kola(cosaDir: env.WORKSPACE, parallel: n,
+             build: newBuildID, arch: basearch,
```
Comment (jlebon) [inline jobs/build-arch.Jenkinsfile:339]: Do we actually need to pass `build` here?

File: jobs/build-arch.Jenkinsfile [hunk 3]
```diff
@@ -446,38 +385,8 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
...
-                    cosa shell -- tar -c --xz tmp/kola-testiso-metal4k/ > kola-testiso-metal4k.tar.xz || :
-                    """)
-                    archiveArtifacts allowEmptyArchive: true, artifacts: 'kola-testiso*.tar.xz'
-                }
-            }
-
-            // reset for the next batch of independent tasks
-            parallelruns = [:]
+            // Run Kola TestISO tests for metal artifacts
+            kolaTestIso(cosaDir: env.WORKSPACE, build: newBuildID, arch: basearch)
```
Comment (jlebon) [inline jobs/build-arch.Jenkinsfile:389]: I don't think `build` does anything for `kolaTestIso`.

File: jobs/build.Jenkinsfile [hunk 4]
```diff
@@ -442,50 +379,18 @@ lock(resource: "build-${params.STREAM}") {
...
-                        }
-                    }
+            // Run Kola TestISO tests for metal artifacts
+            kolaTestIso(cosaDir: env.WORKSPACE, build: newBuildID, arch: basearch)
+            // For now we want to notify ourselves when this workaround is observed. It won't
+            // fail the build, just give us information.
+            // https://github.com/coreos/fedora-coreos-tracker/issues/1233
+            def grepRc = shwrapRc("""
+                 cosa shell -- grep 'tracker issue workaround engaged for .*issues/1233' \
+                    tmp/kola-testiso-uefi*/insecure/{iso-live-login,iso-as-disk}/console.txt
```
Comment (jlebon) [inline jobs/build.Jenkinsfile:389]: I don't think this path is valid anymore.

File: jobs/build.Jenkinsfile [hunk 5]
```diff
@@ -442,50 +379,18 @@ lock(resource: "build-${params.STREAM}") {
...
-                    if (grepRc == 0) {
-                        warnError(message: 'Detected used workaround for #1233') {
-                            error('Detected used workaround for #1233')
-                        }
-                    }
+            // Run Kola TestISO tests for metal artifacts
+            kolaTestIso(cosaDir: env.WORKSPACE, build: newBuildID, arch: basearch)
+            // For now we want to notify ourselves when this workaround is observed. It won't
+            // fail the build, just give us information.
+            // https://github.com/coreos/fedora-coreos-tracker/issues/1233
```
Comment (jlebon) [inline jobs/build.Jenkinsfile:386]: While we're here, let's add a `shwrap` here that just checks that the file we'll grep in the next `shwrapRc` actually exists.

File: jobs/bump-lockfile.Jenkinsfile [hunk 6]
```diff
@@ -206,63 +207,11 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
...
-                                }
-                            } else {
-                                throw e
-                            }
-                        }
-                    }
-                    parallel parallelruns
+                    def n = ncpus - 1 // remove 1 for upgrade test
+                    kola(cosaDir: env.WORKSPACE, parallel: n,
+                         extraArgs: 'ext.config.files.license',
```
Comment (jlebon) [inline jobs/bump-lockfile.Jenkinsfile:212]: Leftover from testing?

File: jobs/build-arch.Jenkinsfile (hunk 2 above)
Comment (dustymabe) [reply jobs/build-arch.Jenkinsfile:339]: No. Resolved

File: jobs/build-arch.Jenkinsfile (hunk 3 above)
Comment (dustymabe) [reply jobs/build-arch.Jenkinsfile:389]: Correct, fixed.

File: jobs/bump-lockfile.Jenkinsfile (hunk 6 above)
Comment (dustymabe) [reply jobs/bump-lockfile.Jenkinsfile:212]: yep - fixed

File: jobs/build-arch.Jenkinsfile (hunk 1 above)
Comment (dustymabe) [reply jobs/build-arch.Jenkinsfile:166]: broken out into a separate commit

File: jobs/build.Jenkinsfile (hunk 4 above)
Comment (dustymabe) [reply jobs/build.Jenkinsfile]: updated the path and comment

File: jobs/build.Jenkinsfile (hunk 5 above)
Comment (dustymabe) [reply jobs/build.Jenkinsfile:385]: done

Comment (dustymabe) [reply jobs/build.Jenkinsfile:385]: From a test run the added `shwrap` shows:

```
19:25:36  + ls tmp/kolaTestIso-BJqWL/kola-testiso-uefi/insecure/iso-live-login/console.txt tmp/kolaTestIso-BJqWL/kola-testiso-uefi/insecure/iso-as-disk/console.txt
19:25:36  tmp/kolaTestIso-BJqWL/kola-testiso-uefi/insecure/iso-as-disk/console.txt
19:25:36  tmp/kolaTestIso-BJqWL/kola-testiso-uefi/insecure/iso-live-login/console.txt
```

File: jobs/build.Jenkinsfile [hunk 7]
```diff
@@ -442,50 +378,22 @@ lock(resource: "build-${params.STREAM}") {
...
-                    }
+            // Run Kola TestISO tests for metal artifacts
+            kolaTestIso(cosaDir: env.WORKSPACE, arch: basearch)
+            // For now we want to notify ourselves when a particular workaround is observed.
+            // It won't fail the build, just give us information.
+            // https://github.com/coreos/fedora-coreos-tracker/issues/1233
+            // XXX: This relies on implementation details in kolatestIso(),
+            //      but since this is a hack and probably short lived that's OK.
+            // First check to make sure the files exist, then grep for the workaround.
+            shwrap("ls tmp/kolaTestIso-*/kola-testiso-uefi/insecure/{iso-live-login,iso-as-disk}/console.txt")
```
Comment (jlebon) [inline jobs/build.Jenkinsfile:389]: Shouldn't this be prefixed by `cosa shell -- `?

**Review by jlebon — APPROVED** [review]
Nice work! One comment, otherwise LGTM.

File: jobs/build.Jenkinsfile (hunk 7 above)
Comment (dustymabe) [reply jobs/build.Jenkinsfile:389]: yes and no :)

**Comment by dustymabe** [comment]
> Nice work!

Thanks!
